City-of-Helsinki/events-helsinki-monorepo
# (wip): Poc common error handler for apollo link in liikunta

## Description
Looks like I finally nailed all the problems with the error pages. 

1) First off, the ErrorBoundary was in a wrong place. Because ErrorBoundary component still needs to show header and footer sections, it needs to be inside the ApolloProvider and CMS Provider, not the outmost part of the _app.
2) 500.tsx didnt work, it always failes to getStaticProps. The reason is a bug in nextjs when middleware.ts is implemented. There was an actually an open issue for that: https://github.com/vercel/next.js/issues/38762
Fortunately this problem can be fixed by simply not using the nextjs autosupported 500-page at all. If the page doesnt exist, then nextjs uses  `_error.tsx` instead.
3) So `_error.tsx` needed to be implemented due to 2) but also because client-side errors (ie. graphql network errors) always defaults to it on nextjs. Next open issue was that the translations were broken on _error page: https://github.com/i18next/next-i18next/issues/1020
There is no fix for that either, so only thing to do was to instantly redirect to some different page, so it will always redirect now to the `error.tsx` which mimics the original 500 page. 

Wohoo, everything works now correctly!

how to test:
- start script to run the apollo server locally `yarn docker:graphql-router:hobbies:serve` from root of the monorepo
- run `yarn build-fast` under `app/sports-helsinki`
- run `yarn start` 
- go to: `http://localhost:3000` and browse around
- stop the apollo server
- browse some more
-> error.tsx page should pop up


## Issues

### Closes

**[DEV-XXX](https://helsinkisolutionoffice.atlassian.net/browse/DEV-XXX):**

### Related

## Testing

### Automated tests

### Manual testing

## Screenshots

## Additional notes
